DBCON-229: Update DB's Doc on how Pooling works within Stored Procedures (#1079)
mulesoft/docs-connectors: db/1.8/modules/ROOT/pages/database-connector-examples.adoc

	</db:stored-procedure>
----

[[use_pooling_profile_stored_procedures]]

> == Use a Pooling Profile on Stored Procedures

Reviewer: this title is not meaningful to what this entry explains, this is how to workaround pooling profile limitations when using DB inside of a iterative flow, not how to use a pooling profile on stored procedures

Reviewer 2: I can add this scenario as a new section for the Troubleshooting doc file.
What about this name **Pooling Profiles Connections Making the Flow Hang**

Author: I agree to move it to the Troubleshooting file, as well as with the new title.

> Using a pooling profilethat provides fewer connections than what the flow uses can make the entire flow hang. One solution is to increase the size of the pool. However, sometimes that's not possible, so another solution is to use the VM Connector to shorten the life of the connection to be the same as the life of a VM message. This way, every call to the *Stored procedure* operation is executed, its result is consumed, and the connection associated with the operation is closed and returned to the pool.

Reviewer 3: ```suggestion
Using a pooling profile that provides fewer connections than what the flow uses can make the entire flow hang. One solution is to increase the size of the pool. However, sometimes that's not possible, so another solution is to use the VM Connector to shorten the life of the connection to be the same as the life of a VM message. This way, every call to the *Stored procedure* operation is executed, its result is consumed, and the connection associated with the operation is closed and returned to the pool.
```

	</db:stored-procedure>
----

[[use_pooling_profile_stored_procedures]]

> == Use Pooling Profile on Stored Procedures

Reviewer 3: ```suggestion
== Use a Pooling Profile on Stored Procedures
```

> Because using a pooling profile provides fewer connections than the connections the flow uses and makes the entire flow hang, a plausible solution is to increase the size of the pool. However, sometimes this is not possible, so another workaround is to use the VM Connector to narrow the life of the connection to be like the life of a VM message. This way every call to the *Stored procedure* operation is executed, its result consumed, and the connection associated with the operation is closed and returned to the pool.

Reviewer 3: ```suggestion
Using a pooling profile that provides fewer connections than what the flow uses can make the entire flow hang. One solution for this is to increase the size of the pool. However, sometimes that's not possible, so another solution is to use the VM Connector to shorten the life of the connection to be the same as the life of a VM message. This way, every call to the *Stored procedure* operation is executed, its result is consumed, and the connection associated with the operation is closed and returned to the pool.
```

> The following example shows how to use pooling profiles with the *Store procedure* operation and VM Connector. When the foreach element reaches the eleventh iteration the *Stored procedure* operation won't hang since the pool is able to provide a connection:

Reviewer 3: ```suggestion
The following example shows how to use pooling profiles with the *Stored procedure* operation and VM Connector. When the `foreach` element reaches the eleventh iteration the *Stored procedure* operation won't hang since the pool is able to provide a connection:
```
